java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/jpacontroller/JpaControllerGenerator.java

```diff
@@ -226,488 +219,501 @@ private static FileObject generateJpaController(
-                    allRelMethods.addAll(toManyRelMethods);
-                    
-                    String[] importFqs = {"javax.persistence.Query",
-                                "javax.persistence.EntityNotFoundException"                              
-                    };
+                }
+                modifiedClassTree = JpaControllerUtil.TreeMakerUtils.addVariable(modifiedClassTree, workingCopy, "emf", "javax.persistence.EntityManagerFactory", privateModifier, null, null);   //NOI18N
+                parameterTypes.add("javax.persistence.EntityManagerFactory");   //NOI18N
+                parameterNames.add("emf");   //NOI18N
+                body += "this.emf = emf;";   //NOI18N
+                MethodInfo mi = new MethodInfo("<init>", publicModifier, "void", null, parameterTypes.toArray(new String[0]),   //NOI18N
+                parameterNames.toArray(new String[0]), body, null, null);
+                modifiedClassTree = JpaControllerUtil.TreeMakerUtils.modifyDefaultConstructor(classTree, modifiedClassTree, workingCopy, mi);
+                
+                MethodInfo methodInfo = new MethodInfo("getEntityManager", publicModifier, "javax.persistence.EntityManager", null, null, null, "return emf.createEntityManager();", null, null);
+                modifiedClassTree = JpaControllerUtil.TreeMakerUtils.addMethod(modifiedClassTree, workingCopy, methodInfo);
+                
+                String bodyText;
+                StringBuilder updateRelatedInCreate = new StringBuilder();
+                StringBuilder updateRelatedInEditPre = new StringBuilder();
+                StringBuilder attachRelatedInEdit = new StringBuilder();
+                StringBuilder updateRelatedInEditPost = new StringBuilder();
+                StringBuilder updateRelatedInDestroy = new StringBuilder();
+                StringBuilder initRelatedInCreate = new StringBuilder();
+                StringBuilder illegalOrphansInCreate = new StringBuilder();
+                StringBuilder illegalOrphansInEdit = new StringBuilder();
+                StringBuilder illegalOrphansInDestroy = new StringBuilder();
+                StringBuilder initCollectionsInCreate = new StringBuilder();  //useful in case user removes listbox from New.jsp
+                
+                List<ElementHandle<ExecutableElement>> allRelMethods = new ArrayList<>(toOneRelMethods);
+                allRelMethods.addAll(toManyRelMethods);
+                
+                String[] importFqs = {"javax.persistence.Query",
+                    "javax.persistence.EntityNotFoundException"
+                };
+                
+                for (String importFq : importFqs) {
+                    modifiedImportCut = JpaControllerUtil.TreeMakerUtils.createImport(workingCopy, modifiedImportCut, importFq);
+                }
+                if(version!=null && !Persistence.VERSION_1_0.equals(version)){//add criteria classes if appropriate
```

Comment:
```suggestion
                if(version != null && !Persistence.VERSION_1_0.equals(version)){//add criteria classes if appropriate
```

Reply:
Thanks for the suggestions, I will create another PR with better formatting, meanwhile lets get this merged.